src/objects/file.py

```diff
@@ -11,93 +12,194 @@
+        self.file_path: str = file_path
+        self.create_binary_from_path()
+
+        #: The virtual file path is not a path on the analysis machine but the full path inside a firmware object.
+        #: For a file inside a filesystem, that was itself packed inside an archive this might look like
+        #: `firmware_uid|fs_uid|/etc/hosts` with the pipe sign ( | ) separating extraction levels.
+        #: For files such as symlinks, there can be multiple paths inside a single firmware for one unique file.
+        self.virtual_file_path: dict = {}
+
+    def set_binary(self, binary: bytes) -> None:
+        '''
+        Store binary of file as byte string.
+        Additionally set binary related meta data (size, hash) and compute uid after that.
+
+        :param binary: file in binary representation
+        '''
         self.binary = make_bytes(binary)
         self.sha256 = get_sha256(self.binary)
         self.size = len(self.binary)
         self._uid = create_uid(binary)
 
-    def set_name(self, name):
-        self.file_name = make_unicode_string(name)
-
-    def set_file_path(self, file_path):
-        self.file_path = file_path
-        if self.binary is None:
-            self.create_from_file(file_path)
-        if self.file_name is None:
-            self.set_name(os.path.basename(file_path))
-
-    def get_uid(self):
-        logging.warning('Deprecation warning: "get_uid()" was replaced by "uid" and will be removed in a future update')
-        return self.uid
+    def create_binary_from_path(self) -> None:
+        if self.file_path is not None:
+            if self.binary is None:
+                self._create_from_file(self.file_path)
+            if self.file_name is None:
+                self.file_name = make_unicode_string(Path(self.file_path).name)
 
     @property
-    def uid(self):
+    def uid(self) -> str:
+        '''
+        Unique identifier of this file.
+        Consisting of the file's sha256 hash and it's length in the form `hash_length`.
+
+        :return: uid of this file.
+        '''
         if self._uid is None and self.binary is not None:
             self._uid = create_uid(self.binary)
         return self._uid
 
     @uid.setter
-    def uid(self, new_uid):
+    def uid(self, new_uid: str):
         if self._uid is not None:
             logging.warning('uid overwrite: Uid might not be related to binary data anymore: {} -> {}'.format(self._uid, new_uid))
         self._uid = new_uid
 
-    def get_hid(self, root_uid=None):
+    def get_hid(self, root_uid: str = None) -> str:
         '''
-        return a human readable identifier
+        Get a human readable identifier for the given file.
+        This usually is the file name for extracted files.
+        As files can have different names across occurences, uid of a specific root object can be specified.
+
+        :param root_uid: (Optional) root uid to base HID on.
+        :return: String presenting human readable identifier for file.
```

Comment:
Why so few articles? 😟
```suggestion
        :return: String representing a human-readable identifier for the file.
```